### PR TITLE
Android x86_64 fix undefined symbol brave::OnBeforeSystemRequest 

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Fixes https://github.com/brave/brave-browser/issues/6277 

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues/6277) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Requires to have https://github.com/brave/brave-core/pull/3417 patches
```
run npm run init -- --target_os=android --target_arch=x64
run npm run build -- --target_os=android --target_arch=x64
```

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: patches/android_webview-BUILD.gn.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+--- a/android_webview/BUILD.gn
++++ b/android_webview/BUILD.gn
+@@ -706,6 +706,9 @@ source_set("common") {
+     "//android_webview/common",
+     "//android_webview/common:common_mojom",
+     "//base/third_party/dynamic_annotations:dynamic_annotations",
++    "//brave/browser/net",
++    "//brave/browser",
++    "//components/translate/core/language_detection",
+     "//components/about_ui",
```
Comment (SergeyZhukovsky) [inline patches/android_webview-BUILD.gn.patch]: make a patch one string only probably, so we patch less.

Comment (AlexeyBarabash) [reply patches/android_webview-BUILD.gn.patch]: @SergeyZhukovsky , do you mean to make this way
```
+ +"//brave/browser/net","//brave/browser","//components/translate/core/language_detection", 
``` 
?

Comment (SergeyZhukovsky) [reply patches/android_webview-BUILD.gn.patch]: yep, the less lines we patch the easier to rebase after that. We don't care about how does it look like as it's a patch in any case.

Comment (bridiver) [reply patches/android_webview-BUILD.gn.patch]: actually what we should do here is create a new target in brave called `//brave/android_webview:common` that includes all 3 of these. That way the patch is extensible and more deps can be added in the future without changing the patch.

Comment (AlexeyBarabash) [reply patches/android_webview-BUILD.gn.patch]: @bridiver @SergeyZhukovsky thanks, minimized patch according to your proposition

Comment (simonhong) [reply patches/android_webview-BUILD.gn.patch]: I think `android_webview` target should not depend on brave layer. It looks like layer violation. WDYT?

Comment (simonhong) [reply patches/android_webview-BUILD.gn.patch]: @jumde is working on this. https://github.com/brave/brave-core/pull/3415#discussion_r333359050

Comment (AlexeyBarabash) [reply patches/android_webview-BUILD.gn.patch]: @simonhong , yes this had confused me , but I could not find any better option.

Comment (AlexeyBarabash) [reply patches/android_webview-BUILD.gn.patch]: ... and I run into `assert_no_deps = [ "//chrome/*" ]` at https://cs.chromium.org/chromium/src/components/BUILD.gn?l=619 when tried to add deps in obvious way

Comment (simonhong) [reply patches/android_webview-BUILD.gn.patch]: Right, I think this link error should be fixed from `safe_browsing_network_context.cc`

**Pull request closed without merging**